doc/hardware/porting/shields.rst

```diff
@@ -17,12 +17,28 @@ under :zephyr_file:`boards/shields`:
 .. code-block:: none
 
    boards/shields/<shield>
+   ├── shield.yml
    ├── <shield>.overlay
    ├── Kconfig.shield
```

Comment:
Many `Kconfig.shield` only contains a line in the form of:
```
config SHIELD_<NAME>
         def_bool $(shields_list_contains,<name>)
```
as we now have `shield.yml` should auto-defined this safe the shield maintainer the typing.

Just like we do for boards:
https://github.com/zephyrproject-rtos/zephyr/blob/ca182edb06a21c6054528a01d7f52e5aaf5cf489/boards/Kconfig.v2#L9-L10

note, we would need some way to be able to handle list of shields.

Reply:
@kartben thoughts on above ^^^ ?

Reply:
Thoughts, yes (i.e. it's a really good idea and we should do it :)) -- but enough knowledge about CMake/the build system to come up with a solution, maybe not :/ I'm afraid I wouldn't know where to start to turn SHIELD_AS_LIST into multiple config options.

Reply:
Fair point, accepted.

Issue for follow-up created: https://github.com/zephyrproject-rtos/zephyr/issues/90209